exchanges/orderbook/linked_list.go

```diff
@@ -340,6 +340,62 @@ func (ll *linkedList) insertUpdates(updts Items, stack *stack, comp comparison)
 	return nil
 }
 
+// getSlippageByVolume returns the slippage percentage by impact volume on
+// liquidity.
+func (ll *linkedList) getSlippageByVolume(volume float64) (float64, error) {
+	var slippage, shiftedPrice float64
+	for tip := &ll.head; *tip != nil; tip = &(*tip).Next {
+		shiftedPrice = (*tip).Value.Price
+		if slippage == 0 {
+			slippage = 1 / shiftedPrice
+		}
+		volume -= (*tip).Value.Amount
+		if volume <= 0 {
+			break
+		}
+	}
+	if shiftedPrice == 0 {
+		return 0, errNoLiquidity
+	}
+	if volume > 0 {
+		// Full book wiped out, return 100 percent.
+		return 100, nil
+	}
+	ratio := (slippage * shiftedPrice) - 1
+	if ratio < 0 { // Return ABS
+		ratio *= -1
+	}
+	return ratio * 100, nil
+}
+
+// getVolumeBySlippage returns the max volume amount by allowable slippage.
+func (ll *linkedList) getVolumeBySlippage(slippage float64) (float64, error) {
+	var volume, initialPrice float64
```

Comment:
Could return on slippage <= 0

Reply:
zero percent slippage is the top head tranche I can filter out negative values.